EIPS/eip-5850.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 5850
+title: Complex Numbers stored in Bytes32 types
+description: Store real and imaginary parts of complex numbers in the lower 16 bytes and upper 16 bytes respectively of a bytes32 type.
+author: Paul Edge (@genkifs)
+discussions-to: https://ethereum-magicians.org/t/TBC
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-10-29
+---
+ 
+## Abstract
+
+This EIP proposes a natural way for complex numbers to be stored in the bytes32 data-type.
+
+## Motivation
+
+This EIP proposes to package both the real and imaginary within one existing data type, Bytes32.  This allows compact storage without the need for structures and facilitates easy library implementations.  The bytes32 would remain available for existing, non-complex uses.
+
+## Specification
```

Comment:
I don't think this EIP works, as it can only store complex numbers with integer parts.

Reply:
> it can only store complex numbers with integer parts

Almost, it can only store numbers that have an *int128* representation.  The problem is that decimal numbers, either fix or floating, are not yet fully supported by Solidity so enforcing similar standards for complex versions is premature.  However, I've now modified the EIP to *suggest* that fixed point methods for *int* types (such as [prb-math](https://github.com/paulrberg/prb-math)) be used with 18 decimal places, matching the *erc-20* decimal suggestion.

Your comment made me rethink the wording.  I've made changes so that both polar and exponential storage are valid as well as the more "natural" cartesian approach.  Keeping the wording open allows this EIP to remain flexible for future EIPs to build upon.